Assets/Scripts/Interactables/Dialogue/NPCDialogue/NPCInteractable.cs

```diff
@@ -70,23 +122,109 @@ public override void OnInteract()
                     startedMission.OnMissionReset -= HandleOnMissionReset;
                     startedMission = null;
                     missionsOffered.RemoveAt(0);
-
-                    conversation = currentMissionNPC.afterConvo;
-
-                    Debug.Log("NPC: afterConvo");
+                    currentMissionConvos = null;
                 }
                 else
                 {
                     Debug.LogError("Should not be getting to here");
                     return;
                 }
             }
-            NPCFacePlayer();
+            if (!conversation.shouldFollow && !conversation.isAutoplayed)
+            {
+                NPCFacePlayer();
+            }
+        }
+
+        if(conversation.isAutoplayed)
+        {
+            TriggerAutoplay();
+        }
+
+        if (conversation.shouldFollow)
+        {
+            TriggerFollow(player);
         }
-        
         base.OnInteract();
     }
 
+    protected override void OnTriggerEnter(Collider other)
+    {
+        if (!autoPlaying)
+        {
+            base.OnTriggerEnter(other);
+        }
+    }
+
+    protected override void OnAutoplayComplete()
+    {
+        base.OnAutoplayComplete();
+        if(conversation.shouldFollow)
+        {
+            StopFollow();
+            ReturnToOrigin();
+        }
+    }
+
+    // Not sure if we'll need this funtionality but putting it here anyway
+    // Would happen when shouldFollow is true and autoPlaying is false for that conversation
+    protected override void EndConversation()
+    {
+        base.EndConversation();
+
+        if(isFollowing && !autoPlaying)
+        {
+            StopFollow();
+            ReturnToOrigin();
+        }
+    }
+
+    void FollowTarget()
+    {
+        if (targetObject == null)
+        {
+            Debug.LogError("Object being followed must be assigned");
+            return;
+        }
+        else
+        {
+            agent.SetDestination(targetObject.transform.position);
+        }
+    }
+
+    private bool IsWithinBoundaryRadius(Transform position)
+    {
+        return Vector3.Distance(originPosition, position.position) < Constants.INTERACT_BOUNDARY_RADIUS;
```

Comment:
`position.position` looks a bit off to me aha, can we rename the parameter to something like `checkTransform` or something?